lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -221,17 +232,6 @@ def _check_host_capabilities(
         total_required_memory_mib = 0
 
         for node_requirements in nodes_requirements:
-            # Ensure host has enough CPU cores for the VM.
-            # Note: The CPU scheduler can easily handle overprovisioning of CPU
-            # cores.
-            assert isinstance(node_requirements.core_count, int)
-            if node_requirements.core_count > host_capabilities.core_count:
```

Comment:
With the change to how node requirements/capabilities are handled, this logic could be moved to `_create_node_capabilities`.